lib/evss/auth_headers.rb

```diff
@@ -39,18 +39,18 @@ def eauth_json
         authorizationResponse: {
           status: 'VETERAN',
           idType: 'SSN',
-          id: @user.ssn,
-          edi: @user.edipi,
-          firstName: @user.first_name,
-          lastName: @user.last_name,
+          id: @user[:ssn],
+          edi: @user[:edipi],
+          firstName: @user[:first_name],
+          lastName: @user[:last_name],
           birthDate: iso8601_birth_date
         }
       }.to_json
     end
 
     def iso8601_birth_date
-      return nil unless @user&.va_profile&.birth_date
-      DateTime.parse(@user.va_profile.birth_date).iso8601
+      return nil unless @user[:va_profile].try(:[],:birth_date)
```

Comment:
Ran into an issue where sometimes the user data sent in is an object and some time's it's a hash because of the constant changing of formats I went ahead and set a .to_h.deep_symbolize_keys! in the initializer in order to make it uniform and made the entire thing hash based so that the state of the user object doesn't matter.